otel-extension/src/it/scala/io/scalac/mesmer/agent/utils/OtelAgentTest.scala

```diff
@@ -1,21 +1,105 @@
 package io.scalac.mesmer.agent.utils
 
-import io.opentelemetry.context.ContextStorage
 import io.opentelemetry.instrumentation.testing.AgentTestRunner
-import io.opentelemetry.instrumentation.testing.util.ContextStorageCloser
-import org.scalatest.BeforeAndAfterAll
-import org.scalatest.BeforeAndAfterEach
-import org.scalatest.TestSuite
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo
+import io.opentelemetry.sdk.metrics.data.{ HistogramPointData, MetricData }
+import io.opentelemetry.sdk.metrics.internal.aggregator.EmptyMetricData
+import io.opentelemetry.sdk.metrics.internal.data.ImmutableMetricData
+import io.opentelemetry.sdk.resources.Resource
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{ BeforeAndAfterAll, BeforeAndAfterEach, OptionValues, TestSuite }
 
-abstract class OtelAgentTest extends TestSuite with BeforeAndAfterAll with BeforeAndAfterEach {
+import scala.jdk.CollectionConverters._
+import scala.util.Try
 
-  private val testRunner = AgentTestRunner.instance()
+trait OtelAgentTest extends TestSuite with BeforeAndAfterAll with Eventually with BeforeAndAfterEach with OptionValues {
+
+  protected val testRunner = AgentTestRunner.instance()
 
   override protected def beforeAll(): Unit = testRunner.beforeTestClass()
 
   override protected def afterAll(): Unit = testRunner.afterTestClass()
 
-  override protected def beforeEach(): Unit = testRunner.clearAllExportedData()
+  override protected def beforeEach(): Unit = {
```

Comment:
So now we have 2 different versions of the otelAgentTest protected methods: 
 - the ones introduced in #418 
 - the ones from this PR

How do we prevent a mixup? Is there a requirement that #418 is merged first, otherwise this version will disappear?

Reply:
Yeah, sorry, this should not be visible right now - this is actually a chained PR - #418 must be merged first